### PR TITLE
[SYCL] Indicate lack of direct OpenCL interop when not using OpenCL BE

### PR DESCRIPTION
**Pull request by rbegam**
Signed-off-by: Sergey V Maslov <sergey.v.maslov@intel.com>

### CODE REVIEW

File: sycl/test/lit.cfg.py [hunk 1]
```diff
@@ -67,6 +67,16 @@
 if 'OCL_ICD_FILENAMES' in os.environ:
     config.environment['OCL_ICD_FILENAMES'] = os.environ['OCL_ICD_FILENAMES']
 
+config.available_features.add('opencl-interop')
+if 'SYCL_BE' in os.environ:
+    config.environment['SYCL_BE'] = os.environ['SYCL_BE']
+    # There is no direct OpenCL interop when not running OpenCL BE.
+    if config.environment['SYCL_BE'] == "PI_OTHER":
```
Comment (bader) [inline sycl/test/lit.cfg.py]:
```suggestion
    if config.environment['SYCL_BE'] != "PI_OPENCL":
```

File: sycl/test/lit.cfg.py [hunk 2]
```diff
@@ -67,6 +67,16 @@
...
+        config.available_features.remove('opencl-interop')
+
+if 'SYCL_DEVICE_WHITE_LIST' in os.environ:
```
Comment (bader) [inline sycl/test/lit.cfg.py]:
```suggestion
if 'SYCL_DEVICE_ALLOWLIST' in os.environ:
```

File: sycl/test/lit.cfg.py [hunk 3]
```diff
@@ -67,6 +67,16 @@
...
+    config.environment['SYCL_DEVICE_WHITE_LIST'] = os.environ['SYCL_DEVICE_WHITE_LIST']
```
Comment (bader) [inline sycl/test/lit.cfg.py]:
```suggestion
    config.environment['SYCL_DEVICE_ALLOWLIST'] = os.environ['SYCL_DEVICE_ALLOWLIST']
```

**Comment by bader** [comment]
@rbegam, please, resolve merge conflicts.

**Comment by rbegam** [comment]
> @rbegam, please, resolve merge conflicts.

@bader Done. Could you please review and approve? Thanks.

**Comment by rbegam** [comment]
> @rbegam, please, resolve merge conflicts.

@bader Done. Could you please review and approve? Thanks.

File: sycl/test/lit.cfg.py [hunk 4]
```diff
@@ -67,6 +67,16 @@
 if 'OCL_ICD_FILENAMES' in os.environ:
     config.environment['OCL_ICD_FILENAMES'] = os.environ['OCL_ICD_FILENAMES']
 
+config.available_features.add('opencl-interop')
+if 'SYCL_BE' in os.environ:
+    config.environment['SYCL_BE'] = os.environ['SYCL_BE']
```
Comment (bader) [inline sycl/test/lit.cfg.py]: I don't this code is aligned with the recently committed CUDA back-end.
This patch assumes that `SYCL_BE` is set in the invironment, where as CUDA back-end patch sets this inside this script.
@Ruyk, @Alexander-Johnston, could you take a look, please?
 I think we should align on the process for running LIT tests on machines supporting multiple back-ends.

File: sycl/test/basic_tests/buffer/buffer.cpp [hunk 5]
```diff
@@ -1,3 +1,5 @@
+// REQUIRES: opencl-interop
+
```
Comment (bader) [inline sycl/test/basic_tests/buffer/buffer.cpp]: Should we remove "XFAIL: cuda" below?
BTW, different tests handle tests like this differently:
- `XFAIL: cuda`
- `UNSUPPOTED: cuda`
- `REQUIRES: opencl`

File: sycl/test/lit.cfg.py (hunk 4 above)
Comment (Ruyk) [reply sycl/test/lit.cfg.py]: @bjoernknafla

**Comment by bjoernknafla** [comment]
Great work to get LIT testing for CUDA pass!

**Comment by bjoernknafla** [comment]
I have more related fixes in the following - not yet quite ready - PR - any feedback would be very appreciated: https://github.com/codeplaysoftware/sycl-for-cuda/pull/17

File: sycl/test/basic_tests/buffer/buffer.cpp [hunk 6]
```diff
@@ -1,3 +1,5 @@
+// REQUIRES: opencl-interop
```
[the remark this answers is not in the document]
Comment (rbegam) [reply sycl/test/basic_tests/buffer/buffer.cpp]: If I understood it correctly, we should keep XFAIL: cuda here. is that right?

File: sycl/test/basic_tests/buffer/buffer_interop.cpp [hunk 7]
```diff
@@ -1,3 +1,5 @@
+// REQUIRES: opencl-interop
```
[the remark this answers is not in the document]
Comment (rbegam) [reply sycl/test/basic_tests/buffer/buffer_interop.cpp]: how about the existing // REQUIRES: opencl at line# 8?

File: sycl/test/basic_tests/buffer/buffer.cpp (hunk 6 above)
[the remark this answers is not in the document]
Comment (bjoernknafla) [reply sycl/test/basic_tests/buffer/buffer.cpp]: Yes. Thank you.

File: sycl/test/lit.cfg.py [hunk 8]
```diff
@@ -67,6 +67,16 @@
...
+    # There is no direct OpenCL interop when not running OpenCL BE.
+    if config.environment['SYCL_BE'] != "PI_OPENCL":
+        config.available_features.remove('opencl-interop')
```
[the remark this answers is not in the document]
Comment (rbegam) [reply sycl/test/lit.cfg.py]: seems like this whole code section became redundant. I am planning to remove this and change all the REQUIRES: opencl-interop -> REQUIRES: opencl. Let me know if that sounds good.

[the remark this answers is not in the document]
Comment (bjoernknafla) [reply sycl/test/lit.cfg.py]: That would be great! Thank you.

**Comment by rbegam** [comment]
@bader @romanovvlad @garimagu @bjoernknafla  
I've submitted a new PR for this change here: https://github.com/intel/llvm/pull/1242
It has all the corrections requested so far. Please review that one instead.

I'll be closing this.

**Pull request closed without merging**